liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/applications/DeployMojoSupport.java

```diff
@@ -239,6 +180,44 @@ protected void installLooseConfigEar(MavenProject proj, LooseConfigData config)
         looseEar.addManifestFile(manifestFile);
     }
 
+    private boolean shouldValidateAppStart() throws MojoExecutionException {
+        try {
+            return new File(serverDirectory.getCanonicalPath()  + "/workarea/.sRunning").exists();
+        } catch (IOException ioe) {
+            throw new MojoExecutionException("Could not get the server directory to determine the state of the server.");
+        }
+    }
+
+    protected void verifyAppStarted(String appFile) throws MojoExecutionException {
+        if (shouldValidateAppStart()) {
+            String appName;
+            if (getAppsDirectory().equals("apps")) {
+                ServerConfigDocument scd = null;
+
+                File serverXML = new File(serverDirectory, "server.xml");
+
+                if (serverXML != null && serverXML.exists()) {
```

Comment:
Can a running server have no server.xml file?

Reply:
No, there should always be one in the server directory.

Reply:
I'll refactor this block.